src/EspressoNitroTEEVerifier.sol

```diff
@@ -58,9 +62,36 @@ contract EspressoNitroTEEVerifier is NitroValidator, IEspressoNitroTEEVerifier,
         address enclaveAddress = address(uint160(uint256(publicKeyHash)));
 
         // Mark the signer as registered
-        if (!registeredSigners[enclaveAddress]) {
-            registeredSigners[enclaveAddress] = true;
-            emit AWSSignerRegistered(enclaveAddress, pcr0Hash);
+        if (!registeredCaffNodes[enclaveAddress]) {
+            registeredCaffNodes[enclaveAddress] = true;
+            emit AWSNitroServiceRegistered(enclaveAddress, pcr0Hash, ServiceType.CaffNode);
+        }
+    }
+    /**
+     * @notice This function registers a new Batch Poster by verifying an attestation from the AWS Nitro Enclave (TEE)
+     * @param attestation The attestation from the AWS Nitro Enclave (TEE)
+     * @param signature The cryptographic signature over the COSESign1 payload (extracted from the attestation)
+     */
+
+    function registerBatchPoster(bytes calldata attestation, bytes calldata signature) external {
+        Ptrs memory ptrs = validateAttestation(attestation, signature);
+        bytes32 pcr0Hash = attestation.keccak(ptrs.pcrs[0]);
+        if (!registeredBatchPosterEnclaveHashes[pcr0Hash]) {
+            revert InvalidAWSEnclaveHash(pcr0Hash);
+        }
+        // The publicKey's first byte 0x04 byte followed which only determine if the public key is compressed or not.
+        // so we ignore the first byte.
+        bytes32 publicKeyHash =
+            attestation.keccak(ptrs.publicKey.start() + 1, ptrs.publicKey.length() - 1);
+
+        // Note: We take the keccak hash first to derive the address.
+        // This is the same which the go ethereum crypto library is doing for PubkeyToAddress()
+        address enclaveAddress = address(uint160(uint256(publicKeyHash)));
+
+        // Mark the signer as registered
+        if (!registeredBatchPosters[enclaveAddress]) {
+            registeredBatchPosters[enclaveAddress] = true;
```

Comment:
can an address be both a caff node and a batch poster? if not, we may need to add a check for this

Reply:
Hmm. I don't think we've discussed this at length, but I don't think we would ever really want that setup tbh. So I think this would be a good check to add. I think I might do it in the next PR I make though. With the way I updated the tests, I would be unable to test registering both the batchposter and the caff node with the data we have for the tests (We haven't been able to generate new test data yet lmao.)